app/src/main/res/navigation/kiwix_nav_graph.xml

```diff
@@ -48,7 +48,10 @@
 
     <action
       android:id="@+id/action_navigation_reader_to_navigation_library"
-      app:destination="@id/libraryFragment" />
+      app:destination="@id/libraryFragment"
+      app:popUpTo="@id/readerFragment"
+      app:popUpToSaveState="true" />
```

Comment:
@MohitMaliDeveloper We should also try to restore  the previous state by adding `restoreState=”true”`.

Reply:
@gouri-panda  Done.